File: program/src/main/java/com/oose/labsafety/laboratory/application/LaboratoryService.java
package com.oose.labsafety.laboratory.application;

import com.oose.labsafety.common.AbstractCatalogService;
import com.oose.labsafety.laboratory.domain.LaboratoryProfile;
import com.oose.labsafety.laboratory.infrastructure.LaboratoryRepository;

public final class LaboratoryService extends AbstractCatalogService<LaboratoryProfile> {

    public LaboratoryService(LaboratoryRepository repository) {
        super(repository);
    }
}

File: src/features/laboratory/service/laboratoryService.js
import { createCrudService } from '../../../shared/service/createCrudService';
import { createLaboratory } from '../domain/Laboratory';
import { laboratoryRepository } from '../infrastructure/laboratoryRepository';

export const laboratoryService = createCrudService({
  repository: laboratoryRepository,
  createEntity: createLaboratory,
  idField: 'labId',
});
